users/api/endpoints.go

```diff
@@ -102,14 +102,50 @@ func listClientsEndpoint(svc users.Service) endpoint.Endpoint {
 	}
 }
 
-func listMembersEndpoint(svc users.Service) endpoint.Endpoint {
+func listMembersByGroupEndpoint(svc users.Service) endpoint.Endpoint {
 	return func(ctx context.Context, request interface{}) (interface{}, error) {
-		req := request.(listMembersReq)
+		req := request.(listMembersByObjectReq)
+		req.objectKind = "groups"
 		if err := req.validate(); err != nil {
 			return memberPageRes{}, errors.Wrap(apiutil.ErrValidation, err)
 		}
 
-		page, err := svc.ListMembers(ctx, req.token, req.groupID, req.Page)
+		page, err := svc.ListMembers(ctx, req.token, req.objectKind, req.objectID, req.Page)
+		if err != nil {
+			return memberPageRes{}, err
+		}
+
+		return buildMembersResponse(page), nil
+	}
+}
+
+func listMembersByChannelEndpoint(svc users.Service) endpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+		req := request.(listMembersByObjectReq)
+		// in spiceDB channel is group kind
```

Comment:
We not having any separate `type` for channel in spiceDB schema, Instead we using `group` type for storing channel relations.   
Here the function name is `listMembersByChannelEndpoint` , which does listing of users in a channel 
This function request we are passing `req.objectKind = "groups"` ,  

This function is related to channels, then Why we here passing `groups` instead of `channels` ? Like this kind of question will raise when someone see the code.     
Answer is same as said in the first, We are not having separate `type` for `channel` in spiceDB schema, Instead we using same `group` type for channel.

Reply:
updated comment